library/secrethub_generate.py

```diff
@@ -95,29 +95,28 @@ def __init__(self):
                 'type': "bool",
             }
         }
-        super(GenerateModule, self).__init__(argument_spec)
+        super(GenerateModule, self).__init__(argument_spec, supports_check_mode=True)
 
     def run(self):
-        try:
-            secret = self.client().generate(
-                self.params.get('path'),
-                self.params.get('length', 0),
-                self.params.get('symbols', False),
-            )
-            self.exit_json(
-                changed=True,
-                secret=secret,
-            )
-        except (CLIInaccessible, GenerateError) as e:
-            self.fail_json(
-                changed=False,
-                msg=str(e),
-            )
-        except ReadError as e:
-            self.fail_json(
-                changed=True,
-                msg=str(e),
-            )
+        if not self.check_mode:
```

Comment:
What happens when the parent directory (or repo, access, etc.) does not exist and you're running this in check mode? Is the expected behavior that it fails or that it reports everything works fine?

Reply:
Good point! Yes, the module should fail in that case. We can use the same solution as I just proposed for the read module: we can do an `ls` on the parent path to check for existence.